### PR TITLE
Sfdk: Check virtualmachine features before modifying properties

### PR DESCRIPTION
**Pull request by vigejolla**

### CODE REVIEW

File: src/tools/sfdk/command.cpp [hunk 1]
```diff
@@ -262,10 +281,16 @@ class VirtualMachinePropertiesAccessor : public PropertiesAccessor
         } else if (name == VM_STORAGE_SIZE_MB) {
             if (!parsePositiveInt(&m_storageSizeMb, value, errorString))
                 return Failed;
-            if (m_storageSizeMb < m_vm->storageSizeMb()) {
+            if ((m_storageSizeMb < m_vm->storageSizeMb())
+                && !(m_vm->features() & VirtualMachine::ShrinkStorageSize)) {
```
Comment (martyone) [inline src/tools/sfdk/command.cpp]: Should be indented once more. I would prefer omitting braces on the left side of `&&`. Similar below.

File: src/tools/sfdk/command.cpp [hunk 2]
```diff
@@ -111,15 +111,26 @@ class PropertiesAccessor
...
+    }
+
     static QString unknownPropertyMessage(const QString &name)
     {
         return tr("Unrecognized property \"%1\"").arg(name);
     }
+
+    static QString modifyNotSupportedMessage(const QString &name)
+    {
+        return tr("Virtual machine does not support modifying property \"%1\"").arg(name);
```
Comment (martyone) [inline src/tools/sfdk/command.cpp]: I would prefer using more general (reusable) message, e.g.

```
static QString readOnlyPropertyMessage()
{
    return tr("Read-only property");
}
```

Could you please also fix the error reporting by adding property name prefix where it is written to stderr in `BuiltinWorker::setProperties`?

```
qerr() << property << ": " << errorString << endl;
```
...and removing the `name` argument from `unknownPropertyMessage`.

(at least I think that would work well after brief check)

File: src/tools/sfdk/command.cpp (hunk 1 above)
Comment (vigejolla) [reply src/tools/sfdk/command.cpp]: fixed

File: src/tools/sfdk/command.cpp (hunk 2 above)
Comment (vigejolla) [reply src/tools/sfdk/command.cpp]: fixed. Seems to work.

**Pull request merged**